### ISSUE TITLE
Decide on strategy to handle key object descriptors

### ISSUE THREAD

**Issue by csiemssen** [csiemssen/APP-RAS-Driving-with-Language#4]
## Necessary Skills of Model
- identify object out of multiple images to given object descriptor
- answer with object descriptor of described object out of multiple images
- ...

## Approaches
- https://arxiv.org/abs/2412.07247 
   -  object center points from descriptors into bounding boxes using [SAM](https://segment-anything.com) 
   - normalise bounding boxes to stitched image grid
  - use inherently possess perception capabilities of internVL to perform grounding detection or reference captioning by using bounding boxes in internVL specific format
  - "Multimodal models like InternVL [CWW+23] inherently possess perception capabilities and can perform grounding detection or reference captioning using bounding boxes in a specific format."
  - "It is important to note that since the large language model predicts bounding box coordinates by predicting the next token, InternVL normalizes all box coordinates to integers between 0 and 1000. Therefore, after image concatenation, we also process the bounding box coordinates accordingly to meet InternVL’s requirements"
  - Match: `~0.47`

- https://opendrivelab.github.io/Challenge%202024/language_CPS.pdf
  - additional training only on key descriptors
  - extract/generate questions regarding descriptors based on train set
  - Extra Pipeline for enhancing prompt:
     - depth information of objects and image with [Depth Anything](https://depth-anything.github.io)
     - isolated query to the VLM to describing and reasoning about the key objects’ states -> added to prompt
        <img width="386" alt="Image" src="https://github.com/user-attachments/assets/ed8746cf-4a41-4447-a4fb-64144894372f" />
  - Match: `~0.93`

- https://opendrivelab.github.io/Challenge%202024/language_ADLM.pdf
  - bring into Lama format *"will give you some objects around the ego car, those objects will be considered for future reasoning.⟨c1, CAM FRONT, 981.7, 561.7⟩ . . ., [QUESTION]"*
  - Complex Optimization of Perception by separate training by multiple steps  (see optimization strategies p.4 - Section 4.3
       - Use [DINO](https://github.com/IDEA-Research/DINO) perform 2D object detection on both the training set and
validation set
       -  "use the training set’s bounding box coordinates and corresponding visual descriptions to fine-tune the Qwen- VL model"
  - Match: `~0.95`

- https://arxiv.org/abs/2312.14150
   - maybe graph structure with representative questions-answers based as context in regarding stages
   - ![Image](https://github.com/user-attachments/assets/1b188537-3d5e-44e9-87ce-cbda5af18c72)
